src/annotator/util/frame.js

```diff
@@ -0,0 +1,25 @@
+/**
+ * Test whether an iframe fills the viewport of an ancestor frame.
+ *
+ * @param {Window} frame
+ * @param {Window} ancestor
+ */
+export function frameFillsAncestor(frame, ancestor) {
+  if (frame === ancestor) {
+    return true;
+  }
+
+  if (frame.parent !== ancestor) {
+    // To keep things simple, we initially only support direct ancestors.
+    return false;
+  }
+
+  if (!frame.frameElement) {
+    // This is a cross-origin iframe. In this case we can't tell if it fills
+    // the parent frame or not.
+    return false;
+  }
+
+  const frameBox = frame.frameElement.getBoundingClientRect();
+  return frameBox.width / frame.parent.innerWidth >= 0.8;
```

Comment:
I had a query about the origin of the value `0.8`. Does it have meaning? Do we need a test for the boundary case?

Reply:
This is an arbitrary threshold. I'll give it a descriptive name. The only boundary case I can immediately see is if `frame.parent.innerWidth` was 0. I don't think that is possible. If it does somehow happen, the function will return `true`. Nothing will crash though.